bot/exts/utilities/color.py

```diff
@@ -0,0 +1,229 @@
+import colorsys
+import json
+import pathlib
+import random
+from io import BytesIO
+
+from PIL import Image, ImageColor
+from discord import Embed, File
+from discord.ext import commands
+from discord.ext.commands.errors import BadArgument
+from rapidfuzz import process
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+THUMBNAIL_SIZE = (80, 80)
+
+
+class Colour(commands.Cog):
+    """Cog for the Colour command."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        with open(pathlib.Path("bot/resources/utilities/ryanzec_colours.json")) as f:
+            self.colour_mapping = json.load(f)
+
+    async def send_colour_response(self, ctx: commands.Context, rgb: tuple[int, int, int]) -> None:
+        """Create and send embed from user given colour information."""
+        r, g, b = rgb
+        name = self._rgb_to_name(rgb)
+        if name == "No match found":
+            name = None
+
+        try:
+            colour_or_color = ctx.invoked_parents[0]
+        except IndexError:
+            colour_or_color = "colour"
+
+        colour_mode = ctx.invoked_with
+        if colour_mode == "random":
+            input_colour = "random"
+        elif colour_mode in ("colour", "color"):
+            input_colour = rgb
+        else:
+            input_colour = ctx.args[2:][0]
+
+        if colour_mode not in ("name", "hex", "random", "color", "colour"):
+            colour_mode = colour_mode.upper()
+        else:
+            colour_mode = colour_mode.title()
+
+        desc = f"{colour_mode} information for `{name or input_colour}`."
```

Comment:
This can be directly used in the embed, the variable is redundant.

Reply:
Just to make the line shorter I pulled it out onto its own line.

Reply:
It's not needed, you can just make the string multiline if it extends the line limit. Variables are used if it acts as a constant or the value is being manipulated/used multiple times in the source.